processing/src/main/java/org/apache/druid/query/topn/Generic1AggPooledTopNScannerPrototype.java

```diff
@@ -54,25 +54,27 @@ public long scanAndAggregate(
   {
     long processedRows = 0;
     int positionToAllocate = 0;
-    while (!cursor.isDoneOrInterrupted()) {
-      final IndexedInts dimValues = dimensionSelector.getRow();
-      final int dimSize = dimValues.size();
-      for (int i = 0; i < dimSize; i++) {
-        int dimIndex = dimValues.get(i);
-        int position = positions[dimIndex];
-        if (position >= 0) {
-          aggregator.aggregate(resultsBuffer, position);
-        } else if (position == TopNAlgorithm.INIT_POSITION_VALUE) {
-          positions[dimIndex] = positionToAllocate;
-          position = positionToAllocate;
-          aggregator.init(resultsBuffer, position);
-          aggregator.aggregate(resultsBuffer, position);
-          positionToAllocate += aggregatorSize;
+    if (granularizer.currentOffsetWithinBucket()) {
```

Comment:
was the `granularizer.currentOffsetWithinBucket()` check added here (& the other prototypes) to enable skipping of empty granular buckets?

Reply:
yea, i did it as an if statement wrapping the loop so we wouldn't need to check `currentOffsetWithinBucket()` twice per loop since the granularizer advance methods also call `currentOffsetWithinBucket`